runtime/lua/vim/lsp/semantic_tokens.lua

```diff
@@ -637,23 +651,58 @@ end
 --- Only has an effect if the buffer is currently active for semantic token
 --- highlighting (|vim.lsp.semantic_tokens.start()| has been called for it)
 ---
----@param bufnr (nil|number) default: current buffer
+---@param bufnr (number|nil) filter by buffer. All buffers if nil, current
+---       buffer if 0
```

Comment:
Do any other LSP APIs behave this way? Usually nil and 0 are interchangeable. I could be wrong.

Reply:
It's definitely unusual. Most API functions use "nil means current buffer." Some exceptions:
- `vim.lsp.codelens.clear()` uses "nil means all buffers".
- `vim.diagnostic.enable()` uses the same "0 is current, nil is all buffers" as this.

For the rule-based approach, I wanted to "refresh tokens in all buffers that have them" on rule changes. That seemed useful enough to add to the API by tweaking the arguments. But if it's too strange/inconsistent, I will revert that change.

Without that change, if you do want to refresh all tokens, you either
- iterate over _all_ buffers and let `force_refresh` ignore the irrelevant ones
- rebuild the list of relevant buffers by iterating over clients, then buffers per client.
Neither seems awful, so again, if "nil means all, 0 means current" is too weird, I'll delete it.

Reply:
That's a good callout. The other cases (vim.lsp.codelens.clear(), vim.diagnostic) are good to know too. 

This seems like a good convention. I will mention it in `develop.txt`